qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,385 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Hamiltonian simulation of tridiagonal Toeplitz symmetric matrices."""
+
+import numpy as np
+from scipy.sparse import diags
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister
+from qiskit.circuit.library import UGate, MCMTVChain
+
+from .linear_system_matrix import LinearSystemMatrix
+
+
+class Tridiagonal(LinearSystemMatrix):
+    """Class of tridiagonal Toeplitz symmetric matrices"""
+
+    def __init__(self, num_state_qubits: int, main_diag: float, off_diag: float,
+                 tolerance: float = 1e-2, evo_time: float = 1.0, trotter: int = 1,
+                 name: str = 'tridi') -> None:
+        """
+        Args:
+            num_state_qubits: the number of qubits where the unitary acts.
+            main_diag: the main diagonal entry
+            off_diag: the off diagonal entry
+            tolerance: the accuracy desired for the approximation
+            evo_time: the time of the Hamiltonian simulation
+            trotter: the number of Trotter steps
+            name: The name of the object.
+        """
+        # define internal parameters
+        self._num_state_qubits = None
+        self._main_entry = None
+        self._off_diag = None
+        self._tolerance = None
+        self._evo_time = None  # makes sure the eigenvalues are contained in [0,1)
+        self._trotter = None
+
+        # store parameters
+        self.main_entry = main_diag
+        self.off_diag = off_diag
+        super().__init__(num_state_qubits=num_state_qubits, tolerance=tolerance, evo_time=evo_time,
+                         name=name)
+        self.trotter = trotter
+
+    @property
+    def num_state_qubits(self) -> int:
+        r"""The number of state qubits representing the state :math:`|x\rangle`.
+
+        Returns:
+            The number of state qubits.
+        """
+        return self._num_state_qubits
+
+    @num_state_qubits.setter
+    def num_state_qubits(self, num_state_qubits: int) -> None:
+        """Set the number of state qubits.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            num_state_qubits: The new number of qubits.
+        """
+        if num_state_qubits != self._num_state_qubits:
+            self._invalidate()
+            self._num_state_qubits = num_state_qubits
+            self._reset_registers(num_state_qubits)
+
+    @property
+    def main_entry(self) -> float:
+        """Return the entry in the main diagonal."""
+        return self._main_entry
+
+    @main_entry.setter
+    def main_entry(self, main_entry: float) -> None:
+        """Set the entry in the main diagonal.
+        Args:
+            main_entry: The new entry in the main diagonal.
+        """
+        self._main_entry = main_entry
+
+    @property
+    def off_diag(self) -> float:
```

Comment:
Do you think we can have more consistent naming between these two. If its `main_entry` then wouldn't it be `off_entry`. And vice versa `off_diag` `main_diag`. A combination name might work too `main_diag_entry` `off_diag_entry`. Maybe this is just me but is it obvious how this entry - a single value, relates to the diagonals,  which presumably comprise multiple values. Does the main one set the main diagonal entries in the matrix to that value etc?

Reply:
this class implements tridiagonal Toeplitz matrices which have one value, a, in all the main diagonal entries and another value, b, in all the off-diagonal entries. Everything else is 0, so maybe it would make sense to rename the class to `tridiagonal_toeplitz`.

Reply:
If thats more descriptive its fine by me. I noted below, when I noticed later, that the param names in the constructor are main_diag and off_diag and so are consistent there and that there is therefore a mismatch the with the property naming (ie main_entry that returns main_diag, while off_diag stays the same name as one would expect)

Reply:
Renamed the file, also both main and off are now named with the suffix _diag for consistency.